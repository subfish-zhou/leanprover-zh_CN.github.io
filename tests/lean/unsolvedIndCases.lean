theorem ex1 (x : Nat) : 0 + x = x := by
  cases x with
  | zero   => skip -- Error: unsolved goals
  | succ y => skip -- Error: unsolved goals

theorem ex2 (x : Nat) : 0 + x = x := by
  induction x with
  | zero      => skip -- Error: unsolved goals
  | succ y ih => skip -- Error: unsolved goals

theorem ex3 (x : Nat) : 0 + x = x := by
  cases x with
  | zero   => rfl
  | succ y => skip -- Error: unsolved goals

theorem ex4 (x : Nat) {y : Nat} (h : y > 0) : x % y < y := by
  induction x, y using Nat.mod.inductionOn with
  | ind x y h₁ ih => skip -- Error: unsolved goals
  | base x y h₁   => skip -- Error: unsolved goals

theorem ex5 (x : Nat) {y : Nat} (h : y > 0) : x % y < y := by
  cases x, y using Nat.mod.inductionOn with
  | ind x y h₁ ih => skip -- Error: unsolved goals
  | base x y h₁   => skip -- Error: unsolved goals
