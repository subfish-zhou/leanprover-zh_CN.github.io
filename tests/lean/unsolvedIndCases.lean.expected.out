unsolvedIndCases.lean:3:11-3:18: error: unsolved goals
case zero
⊢ 0 + Nat.zero = Nat.zero
unsolvedIndCases.lean:4:11-4:18: error: unsolved goals
case succ
y : Nat
⊢ 0 + Nat.succ y = Nat.succ y
unsolvedIndCases.lean:8:14-8:21: error: unsolved goals
case zero
⊢ 0 + Nat.zero = Nat.zero
unsolvedIndCases.lean:9:14-9:21: error: unsolved goals
case succ
y : Nat
ih : 0 + y = y
⊢ 0 + Nat.succ y = Nat.succ y
unsolvedIndCases.lean:14:11-14:18: error: unsolved goals
case succ
y : Nat
⊢ 0 + Nat.succ y = Nat.succ y
unsolvedIndCases.lean:18:18-18:25: error: unsolved goals
case ind
x y : Nat
h₁ : 0 < y ∧ y ≤ x
ih : y > 0 → (x - y) % y < y
h : y > 0
⊢ x % y < y
unsolvedIndCases.lean:19:18-19:25: error: unsolved goals
case base
x y : Nat
h₁ : ¬(0 < y ∧ y ≤ x)
h : y > 0
⊢ x % y < y
unsolvedIndCases.lean:23:18-23:25: error: unsolved goals
case ind
x y : Nat
h : y > 0
h₁ : 0 < y ∧ y ≤ x
ih : x = x - y → y = y → x % y < y
⊢ x % y < y
unsolvedIndCases.lean:24:18-24:25: error: unsolved goals
case base
x y : Nat
h : y > 0
h₁ : ¬(0 < y ∧ y ≤ x)
⊢ x % y < y
